### ISSUE TITLE
[api-documenter] Type unions are not escaped in markdown

### ISSUE THREAD

**Issue by CarstenLeue** [Microsoft/web-build-tools#1065]
I have this definition:
```typescript
/** Some class doc */
export class MyClass {

  /**
   * Doc of my exported function
   *
   * @param aParam - first parameter
   */
  myFunctionWithUnion(aParam: string | number) {}
}
```
that uses a union type as a parameter.

The generated markdown by the api-documenter (version 7.0.21) does not escape the pipe symbol, which breaks the surrounding table.

Markdown is:
```markdown
## Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  aParam | `string | number` | first parameter |

<b>Returns:</b>

`void`
```

but it should be 

```markdown
## Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  aParam | `string \| number` | first parameter |

<b>Returns:</b>

`void`
```

Without the escaping the word "number" appears as the description and the description is not displayed at all.

**Comment by octogonz** [comment]
Ahhh!  [head explodes]

This is really a bug with GitHub itself:  CommonMark says that ["Note that backslash escapes do not work in code spans. All backslashes are treated literally"](https://spec.commonmark.org/0.28/#code-spans).  And the GFB spec also says ["Backslash escapes do not work in code blocks, code spans, autolinks, or raw HTML"](https://github.github.com/gfm/).  Other markdown engines such as [markdown-it](https://markdown-it.github.io/) (used by Visual Studio) will (correctly) render the backslash inside `string \| number` even when inside a pipes-and-dashes table.

I suppose ultimately **api-documenter** should have an option to customize the markdown flavor that it emits, but we were trying to keep it simple and generic.

**Comment by octogonz** [comment]
This seems to render correctly in both GitHub and **markdown-it**:

```markdown
## Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  aParam | <code>string &#124; number</code> | first parameter |

<b>Returns:</b>

`void`
```

Maybe we could do something like that

**Comment by CarstenLeue** [comment]
@octogonz that would be cool. What happens if any of the types contains a `<` or `>`  character, e.g. because it is a template type. Will these characters have to be escaped?

Example:
```markdown
## Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  aParam | <code>string &#124; MyClass<MyType></code> | first parameter |

<b>Returns:</b>

`void`
```

**Comment by octogonz** [comment]
For example:

```html
<h2>Parameters</h2>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>aParam</td>
<td><code>string | number</code></td>
<td>first parameter</td>
</tr>
</tbody>
</table>
<p><b>Returns:</b></p>
<p><code>void</code></p>
```

(By "HTML" I'm referring to the subset of HTML used by Markdown to describe content. Not full HTML that specifies the visual presentation of the web site.)

**Comment by CarstenLeue** [comment]
@octogonz exporting html would certainly be an option, but as long as markdown output is supported (and I think  that this would be desirable), then the output should be syntactically correct markdown.

**Comment by acchou** [comment]
I'm also hitting this issue. Union types are pretty common in TypeScript. Either the `<code>` solution or html output would work for my use case.

**Comment by octogonz** [comment]
Seems like the portable solution is to emit code spans as `<code>` elements when they appear inside a pipes-and-dashes table.  And as @CarstenLeue pointed out, this requires escaping symbol characters by replacing them with HTML-entities.

Should be pretty easy fix.

**Comment by acchou** [comment]
I'll take a stab at this.

**Comment by iclanton** [comment]
The fix for this has been released in api-documenter version 7.0.41
